### ISSUE TITLE
Error executing 'postInstallation': mysqldump: Error: Binlogging on server not active

### ISSUE THREAD

**Issue by joseairosa** [bitnami/bitnami-docker-mysql#38]
**Description**

When running `docker-compose up` for the `replication` config I get an error on the slave instance.

`Error executing 'postInstallation': mysqldump: Error: Binlogging on server not active`

**Steps to reproduce the issue:**

1. Run `docker-compose --file docker-compose-replication.yml up`

**Describe the results you received:**

Getting the following logs:

```
mysql-master_1  | 2017-06-15T10:50:02.826918Z 0 [Warning] 'user' entry 'mysql.sys@localhost' ignored in --skip-name-resolve mode.
mysql-master_1  | 2017-06-15T10:50:02.826959Z 0 [Warning] 'db' entry 'sys mysql.sys@localhost' ignored in --skip-name-resolve mode.
mysql-master_1  | 2017-06-15T10:50:02.826980Z 0 [Warning] 'proxies_priv' entry '@ root@localhost' ignored in --skip-name-resolve mode.
mysql-master_1  | 2017-06-15T10:50:02.827959Z 0 [Warning] 'tables_priv' entry 'sys_config mysql.sys@localhost' ignored in --skip-name-resolve mode.
mysql-master_1  | 2017-06-15T10:50:02.833592Z 0 [Note] Event Scheduler: Loaded 0 events
mysql-master_1  | 2017-06-15T10:50:02.833885Z 0 [Note] /opt/bitnami/mysql/bin/mysqld: ready for connections.
mysql-master_1  | Version: '5.7.18'  socket: '/opt/bitnami/mysql/tmp/mysql.sock'  port: 3306  MySQL Community Server (GPL)
mysql-master_1  | 2017-06-15T10:50:02.833909Z 0 [Note] Executing 'SELECT * FROM INFORMATION_SCHEMA.TABLES;' to get a list of tables using the deprecated partition engine. You may use the startup option '--disable-partition-engine-check' to skip this check.
mysql-master_1  | 2017-06-15T10:50:02.833913Z 0 [Note] Beginning of list of non-natively partitioned tables
mysql-master_1  | 2017-06-15T10:50:02.855000Z 0 [Note] End of list of non-natively partitioned tables
mysql-slave_1   | mysql   INFO  ==> Configuring replication mode slave...
mysql-slave_1   | mysql   INFO  ==> Setting up database slave...
mysql-slave_1   | mysql   INFO  ==> Checking if replication master is ready to accept connection ...
mysql-slave_1   | INFO  Trying to connect to MySQL server
mysql-slave_1   | INFO  Found MySQL server listening at mysql-master:3306
mysql-slave_1   | INFO  MySQL server listening and working at mysql-master:3306
mysql-slave_1   | mysql   INFO  ==> Setting the master configuration...
mysql-slave_1   | mysql   INFO  ==> Creating a data snapshot...
mysql-slave_1   | Error executing 'postInstallation': mysqldump: Error: Binlogging on server not active
mysql-slave_1   |
bitnamidockermysql_mysql-slave_1 exited with code 1
```

**Describe the results you expected:**

Expected to not see an error

**Version**

- Output of `docker version`:

```
Client:
 Version:      17.03.1-ce
 API version:  1.27
 Go version:   go1.7.5
 Git commit:   c6d412e
 Built:        Tue Mar 28 00:40:02 2017
 OS/Arch:      darwin/amd64

Server:
 Version:      17.03.1-ce
 API version:  1.27 (minimum version 1.12)
 Go version:   go1.7.5
 Git commit:   c6d412e
 Built:        Fri Mar 24 00:00:50 2017
 OS/Arch:      linux/amd64
 Experimental: true
```

- Output of `docker info`:

```
Containers: 16
 Running: 0
 Paused: 0
 Stopped: 16
Images: 64
Server Version: 17.03.1-ce
Storage Driver: aufs
 Root Dir: /var/lib/docker/aufs
 Backing Filesystem: extfs
 Dirs: 224
 Dirperm1 Supported: true
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins:
 Volume: local
 Network: bridge host ipvlan macvlan null overlay
Swarm: inactive
Runtimes: runc
Default Runtime: runc
Init Binary: docker-init
containerd version: 4ab9917febca54791c5f071a9d1f404867857fcc
runc version: 54296cf40ad8143b62dbcaa1d90e520a2136ddfe
init version: 949e6fa
Security Options:
 seccomp
  Profile: default
Kernel Version: 4.9.27-moby
Operating System: Alpine Linux v3.5
OSType: linux
Architecture: x86_64
CPUs: 4
Total Memory: 1.952 GiB
Name: moby
ID: KE5W:3VTM:EKSS:HFG6:E2B6:52RQ:HAZI:TZX6:EUML:4OCC:V2UL:3AFX
Docker Root Dir: /var/lib/docker
Debug Mode (client): false
Debug Mode (server): true
 File Descriptors: 21
 Goroutines: 28
 System Time: 2017-06-15T11:00:00.163040361Z
 EventsListeners: 1
No Proxy: *.local, 169.254/16
Username: joseairosa
Registry: https://index.docker.io/v1/
Experimental: true
Insecure Registries:
 127.0.0.0/8
Live Restore Enabled: false
```

- Output of `docker-compose version` (if applicable):

```
docker-compose version 1.11.2, build dfed245
docker-py version: 2.1.0
CPython version: 2.7.12
OpenSSL version: OpenSSL 1.0.2j  26 Sep 2016
```

**Additional environment details (AWS, VirtualBox, Docker for MAC, physical, etc.):**

Running this on my local machine. Mac OS X Yosemite

**Comment by sameersbn** [comment]
Hello @joseairosa, unfortunately I am not able to reproduce the issue.

Here are the steps I followed

```bash
$ curl https://raw.githubusercontent.com/bitnami/bitnami-docker-mysql/master/docker-compose-replication.yml > docker-compose.yml
$ docker-compose pull
$ docker-compose up
```

The output of `docker-compose up` is pasted below.

Could you please make sure you are using the most recent releases of the `bitnami/mysql` image using the `docker-compose pull` command and then retry?

```console
Sameers-MBP:bitnami sameersbn$ docker-compose up
Creating network "bitnami_default" with the default driver
Creating volume "bitnami_mysql_master_data" with local driver
Creating bitnami_mysql-master_1
Creating bitnami_mysql-slave_1
Attaching to bitnami_mysql-master_1, bitnami_mysql-slave_1
mysql-master_1  |
mysql-master_1  | Welcome to the Bitnami mysql container
mysql-slave_1   |
mysql-slave_1   | Welcome to the Bitnami mysql container
mysql-slave_1   | Subscribe to project updates by watching https://github.com/bitnami/bitnami-docker-mysql
mysql-master_1  | Subscribe to project updates by watching https://github.com/bitnami/bitnami-docker-mysql
mysql-slave_1   | Submit issues and feature requests at https://github.com/bitnami/bitnami-docker-mysql/issues
mysql-master_1  | Submit issues and feature requests at https://github.com/bitnami/bitnami-docker-mysql/issues
mysql-slave_1   | Send us your feedback at containers@bitnami.com
mysql-master_1  | Send us your feedback at containers@bitnami.com
mysql-master_1  |
mysql-slave_1   |
mysql-master_1  | WARN  ==> You set the environment variable ALLOW_EMPTY_PASSWORD=yes. For safety reasons, do not use this flag in a production environment.
mysql-slave_1   | WARN  ==> You set the environment variable ALLOW_EMPTY_PASSWORD=yes. For safety reasons, do not use this flag in a production environment.
mysql-master_1  | nami    INFO  Initializing mysql
mysql-master_1  | mysql   INFO  ==> Configuring permissions...
mysql-master_1  | mysql   INFO  ==> Validating inputs...
mysql-master_1  | mysql   WARN  Allowing the "rootPassword" input to be empty
mysql-master_1  | mysql   WARN  Allowing the "password" input to be empty
mysql-master_1  | mysql   WARN  Allowing the "replicationPassword" input to be empty
mysql-master_1  | mysql   INFO  ==> Initializing database...
mysql-slave_1   | nami    INFO  Initializing mysql
mysql-slave_1   | mysql   INFO  ==> Configuring permissions...
mysql-slave_1   | mysql   INFO  ==> Validating inputs...
mysql-slave_1   | mysql   WARN  Allowing the "replicationPassword" input to be empty
mysql-slave_1   | mysql   INFO  ==> Initializing database...
mysql-master_1  | mysql   INFO  ==> Creating 'root' user with unrestricted access...
mysql-master_1  | mysql   INFO  ==> Creating database my_database...
mysql-master_1  | mysql   INFO  ==> Creating user my_user...
mysql-master_1  | mysql   INFO  ==> Granting access to my_user to the database my_database...
mysql-master_1  | mysql   INFO  ==> Configuring replication mode master...
mysql-master_1  | mysql   INFO  ==> Creating replication user repl_user...
mysql-slave_1   | mysql   INFO  ==> Configuring replication mode slave...
mysql-slave_1   | mysql   INFO  ==> Setting up database slave...
mysql-slave_1   | mysql   INFO  ==> Checking if replication master is ready to accept connection ...
mysql-slave_1   | INFO  Trying to connect to MySQL server
mysql-master_1  | mysql   INFO  ==> Enabling remote connections...
mysql-master_1  | mysql   INFO
mysql-master_1  | mysql   INFO  ########################################################################
mysql-master_1  | mysql   INFO   Installation parameters for mysql:
mysql-master_1  | mysql   INFO     Root User: root
mysql-master_1  | mysql   INFO     Root Password: Not set during installation
mysql-master_1  | mysql   INFO     Database: my_database
mysql-master_1  | mysql   INFO     Username: my_user
mysql-master_1  | mysql   INFO     Password: Not set during installation
mysql-master_1  | mysql   INFO     Replication Mode: master
mysql-master_1  | mysql   INFO     Replication Username: repl_user
mysql-master_1  | mysql   INFO     Replication Password: Not set during installation
mysql-master_1  | mysql   INFO   (Passwords are not shown for security reasons)
mysql-master_1  | mysql   INFO  ########################################################################
mysql-master_1  | mysql   INFO
mysql-master_1  | nami    INFO  mysql successfully initialized
mysql-master_1  | INFO  ==> Starting mysql...
mysql-master_1  | 2017-06-19T10:14:47.273659Z 0 [Warning] 'user' entry 'mysql.sys@localhost' ignored in --skip-name-resolve mode.
mysql-master_1  | 2017-06-19T10:14:47.273718Z 0 [Warning] 'db' entry 'sys mysql.sys@localhost' ignored in --skip-name-resolve mode.
mysql-master_1  | 2017-06-19T10:14:47.273731Z 0 [Warning] 'proxies_priv' entry '@ root@localhost' ignored in --skip-name-resolve mode.
mysql-master_1  | 2017-06-19T10:14:47.274602Z 0 [Warning] 'tables_priv' entry 'sys_config mysql.sys@localhost' ignored in --skip-name-resolve mode.
mysql-master_1  | 2017-06-19T10:14:47.280170Z 0 [Note] Event Scheduler: Loaded 0 events
mysql-master_1  | 2017-06-19T10:14:47.280449Z 0 [Note] /opt/bitnami/mysql/bin/mysqld: ready for connections.
mysql-master_1  | Version: '5.7.18-log'  socket: '/opt/bitnami/mysql/tmp/mysql.sock'  port: 3306  MySQL Community Server (GPL)
mysql-master_1  | 2017-06-19T10:14:47.280475Z 0 [Note] Executing 'SELECT * FROM INFORMATION_SCHEMA.TABLES;' to get a list of tables using the deprecated partition engine. You may use the startup option '--disable-partition-engine-check' to skip this check.
mysql-master_1  | 2017-06-19T10:14:47.280479Z 0 [Note] Beginning of list of non-natively partitioned tables
mysql-master_1  | 2017-06-19T10:14:47.287237Z 0 [Note] End of list of non-natively partitioned tables
mysql-slave_1   | INFO  Found MySQL server listening at mysql-master:3306
mysql-slave_1   | INFO  MySQL server listening and working at mysql-master:3306
mysql-slave_1   | mysql   INFO  ==> Setting the master configuration...
mysql-slave_1   | mysql   INFO  ==> Creating a data snapshot...
mysql-slave_1   | mysql   INFO  ==> Enabling remote connections...
mysql-slave_1   | mysql   INFO
mysql-slave_1   | mysql   INFO  ########################################################################
mysql-slave_1   | mysql   INFO   Installation parameters for mysql:
mysql-slave_1   | mysql   INFO     Root User: root
mysql-slave_1   | mysql   INFO     Root Password: Not set during installation
mysql-slave_1   | mysql   INFO     Database: my_database
mysql-slave_1   | mysql   INFO     Username: my_user
mysql-slave_1   | mysql   INFO     Password: Not set during installation
mysql-slave_1   | mysql   INFO     Replication Mode: slave
mysql-slave_1   | mysql   INFO     Replication Username: repl_user
mysql-slave_1   | mysql   INFO     Replication Password: Not set during installation
mysql-slave_1   | mysql   INFO     Master Host: mysql-master
mysql-slave_1   | mysql   INFO     Master Port: 3306
mysql-slave_1   | mysql   INFO     Master Root User: root
mysql-slave_1   | mysql   INFO     Master Root Password: Not set during installation
mysql-slave_1   | mysql   INFO   (Passwords are not shown for security reasons)
mysql-slave_1   | mysql   INFO  ########################################################################
mysql-slave_1   | mysql   INFO
mysql-slave_1   | nami    INFO  mysql successfully initialized
mysql-slave_1   | INFO  ==> Starting mysql...
mysql-master_1  | 2017-06-19T10:14:57.941932Z 6 [Note] Start binlog_dump to master_thread_id(6) slave_server(821), pos(, 4)
mysql-slave_1   | 2017-06-19T10:14:57.946613Z 0 [Note] /opt/bitnami/mysql/bin/mysqld: ready for connections.
mysql-slave_1   | Version: '5.7.18-log'  socket: '/opt/bitnami/mysql/tmp/mysql.sock'  port: 3306  MySQL Community Server (GPL)
mysql-slave_1   | 2017-06-19T10:14:57.946651Z 0 [Note] Executing 'SELECT * FROM INFORMATION_SCHEMA.TABLES;' to get a list of tables using the deprecated partition engine. You may use the startup option '--disable-partition-engine-check' to skip this check.
mysql-slave_1   | 2017-06-19T10:14:57.946656Z 0 [Note] Beginning of list of non-natively partitioned tables
mysql-slave_1   | 2017-06-19T10:14:57.955001Z 0 [Note] End of list of non-natively partitioned tables
mysql-slave_1   | 2017-06-19T10:14:58.074596Z 2 [Note] Slave SQL thread for channel '' initialized, starting replication in log 'FIRST' at position 0, relay log './mysql-relay-bin.000001' position: 4
mysql-slave_1   | 2017-06-19T10:14:58.705606Z 2 [Warning] 'user' entry 'mysql.sys@localhost' ignored in --skip-name-resolve mode.
mysql-slave_1   | 2017-06-19T10:14:58.705660Z 2 [Warning] 'db' entry 'sys mysql.sys@localhost' ignored in --skip-name-resolve mode.
mysql-slave_1   | 2017-06-19T10:14:58.705680Z 2 [Warning] 'proxies_priv' entry '@ root@localhost' ignored in --skip-name-resolve mode.
mysql-slave_1   | 2017-06-19T10:14:58.705745Z 2 [Warning] 'tables_priv' entry 'sys_config mysql.sys@localhost' ignored in --skip-name-resolve mode.
```

Environment Details:

```
Host OS: MacOS Sierra (10.12.5)
Platform: Docker for Mac 
Version 17.03.1-ce-mac12 (17661)
Channel: stable
```

```bash
$ docker version
Client:
 Version:      17.03.1-ce
 API version:  1.27
 Go version:   go1.7.5
 Git commit:   c6d412e
 Built:        Tue Mar 28 00:40:02 2017
 OS/Arch:      darwin/amd64

Server:
 Version:      17.03.1-ce
 API version:  1.27 (minimum version 1.12)
 Go version:   go1.7.5
 Git commit:   c6d412e
 Built:        Fri Mar 24 00:00:50 2017
 OS/Arch:      linux/amd64
 Experimental: true
```

```bash
$ docker info
Containers: 2
 Running: 0
 Paused: 0
 Stopped: 2
Images: 35
Server Version: 17.03.1-ce
Storage Driver: overlay2
 Backing Filesystem: extfs
 Supports d_type: true
 Native Overlay Diff: true
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins:
 Volume: local
 Network: bridge host ipvlan macvlan null overlay
Swarm: inactive
Runtimes: runc
Default Runtime: runc
Init Binary: docker-init
containerd version: 4ab9917febca54791c5f071a9d1f404867857fcc
runc version: 54296cf40ad8143b62dbcaa1d90e520a2136ddfe
init version: 949e6fa
Security Options:
 seccomp
  Profile: default
Kernel Version: 4.9.27-moby
Operating System: Alpine Linux v3.5
OSType: linux
Architecture: x86_64
CPUs: 4
Total Memory: 1.952 GiB
Name: moby
ID: WDKB:JC3I:AICJ:CHRM:53ZJ:I3SZ:65K7:CO55:KVWD:23DQ:VQBF:LVW3
Docker Root Dir: /var/lib/docker
Debug Mode (client): false
Debug Mode (server): true
 File Descriptors: 17
 Goroutines: 28
 System Time: 2017-06-19T10:36:11.568654443Z
 EventsListeners: 1
No Proxy: *.local, 169.254/16
Registry: https://index.docker.io/v1/
Experimental: true
Insecure Registries:
 127.0.0.0/8
Live Restore Enabled: false
```

**Comment by sameersbn** [comment]
update: i'm able to reproduce the issue with a `docker-compose down ; docker-compose up` sequence. i'm investigating the issue and will get back asap

**Comment by joseairosa** [comment]
Awesome! Thanks, @sameersbn.

Let me know if I can help with anything.

**Comment by sameersbn** [comment]
@joseairosa While we are trying to resolve the issue, I would like to divert your attention to the `bitnami/mariadb` image (compatible with existing mysql clients) which does not have this issue.

**Comment by joseairosa** [comment]
@sameersbn perfect! I'll try that out 👍

**Comment by sameersbn** [comment]
@joseairosa Please note that the issue has been resolved in the `bitnami/mysql:5.7.18-r6` release. Please update your `docker-compose.yml` file and don't forget to do a `docker-compose pull` before retrying.
 
Thanks for taking the time to report this issue. Closing now, re-open of problem persists.
